Plugins/Public/base_plugin/Main.cpp

```diff
@@ -2021,92 +2021,79 @@ void BaseDestroyed(uint space_obj, uint client)
-					iDmgToSpaceID, dmg->get_inflictor_id(), curr, max, damage, dmg->get_cause(), dmg->is_inflictor_a_player(), dmg->get_inflictor_owner_player(), fate);
-
-			if (set_holiday_mode)
-			{
-				returncode = SKIPPLUGINS_NOFUNCTIONCALL;
-				iDmgToSpaceID = 0;
-				return;
-			}
-
-			// A work around for an apparent bug where mines/missiles at the base
-			// causes the base damage to jump down to 0 even if the base is
-			// otherwise healthy.
-			if (damage == 0.0f /*&& dmg->get_cause()==7*/ && curr > 200000)
-			{
-				returncode = SKIPPLUGINS_NOFUNCTIONCALL;
-				if (set_plugin_debug)
-					ConPrint(L"HkCb_AddDmgEntry[1] - invalid damage?\n");
-				return;
-			}
-
-			// If this is an NPC hit then suppress the call completely
-			if (!dmg->is_inflictor_a_player())
-			{
-				if (set_plugin_debug)
-					ConPrint(L"HkCb_AddDmgEntry[2] suppressed - npc\n");
-				returncode = SKIPPLUGINS_NOFUNCTIONCALL;
-				iDmgToSpaceID = 0;
-				return;
-			}
+        return;
+    }
+    
+	if (!spaceobj_modules.count(iDmgToSpaceID)) {
+		return;
+	}
 
-			// Ask the combat magic plugin if we need to do anything differently
-			COMBAT_DAMAGE_OVERRIDE_STRUCT info;
-			info.iMunitionID = iDmgMunitionID;
-			info.fDamageMultiplier = 0.0f;
-			Plugin_Communication(COMBAT_DAMAGE_OVERRIDE, &info);
+    Module* damagedModule = spaceobj_modules[iDmgToSpaceID];
+    if(damagedModule->mining){
+        return;
+    }
+    
+    if (set_holiday_mode)
+	{
+		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+		iDmgToSpaceID = 0;
+		return;
+	}
+	float curr, max;
+	pub::SpaceObj::GetHealth(iDmgToSpaceID, curr, max);
 
-			if (info.fDamageMultiplier != 0.0f)
-			{
-				//ConPrint(L"base: Got a response back, info.fDamage = %0.0f\n", info.fDamage);
-				//ConPrint(L"base: Got a response back, changing damage = %0.0f -> ", damage);
-				damage = (curr - (curr - damage) * info.fDamageMultiplier);
-				if (damage < 0.0f)
-					damage = 0.0f;
-				//ConPrint(L"%0.0f\n", damage);
-			}
+	if (set_plugin_debug)
+		ConPrint(L"HkCb_AddDmgEntry iDmgToSpaceID=%u get_inflictor_id=%u curr=%0.2f max=%0.0f newHealth=%0.2f cause=%u is_player=%u player_id=%u fate=%u\n",
+			iDmgToSpaceID, dmg->get_inflictor_id(), curr, max, newHealth, dmg->get_cause(), dmg->is_inflictor_a_player(), dmg->get_inflictor_owner_player(), fate);
 
-			// This call is for us, skip all plugins.		
-			float new_damage = i->second->SpaceObjDamaged(iDmgToSpaceID, dmg->get_inflictor_id(), curr, damage);
-			returncode = SKIPPLUGINS;
+	// A work around for an apparent bug where mines/missiles at the base
+	// causes the base damage to jump down to 0 even if the base is
+	// otherwise healthy.
+	if (newHealth == 0.0f /*&& dmg->get_cause()==7*/ && curr > 200000)
+	{
+		returncode = SKIPPLUGINS_NOFUNCTIONCALL;
+		if (set_plugin_debug)
+			ConPrint(L"HkCb_AddDmgEntry[1] - invalid damage?\n");
+		return;
+	}
 
-			if (new_damage == 0.0f)
```

Comment:
gone?

Reply:
new_damage variable no longer exists, since we're passing newHealth as reference, and have no need to invoke the separate dmg entry via `dmg->add_damage_entry(p1, new_damage, fate);`